plugin/firewall/policy/expression.go

```diff
@@ -0,0 +1,120 @@
+package policy
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/coredns/coredns/plugin/pkg/rqdata"
+
+	"github.com/coredns/coredns/request"
+
+	"github.com/coredns/coredns/plugin/metadata"
+
+	expr "github.com/Knetic/govaluate"
+)
+
+type ruleExpr struct {
+	kind        int
```

Comment:
I don't understand these `kind`s here. And also not the one below.

Reply:
agree. will replace the `kind` by `action`. It represent the "policy action" associated with the expression.